Tests/Unit/Tools/SmokeTests/QueueDriverConnectionSmokeTestTest.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace Smartbox\Integration\FrameworkBundle\Tests\Unit\Tools\SmokeTests;
+
+use PhpAmqpLib\Exception\AMQPProtocolException;
+use PHPUnit\Framework\TestCase;
+use Smartbox\CoreBundle\Utils\SmokeTest\Output\SmokeTestOutput;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\Drivers\QueueDriverInterface;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessageInterface;
+use Smartbox\Integration\FrameworkBundle\Tools\SmokeTests\QueueDriverConnectionSmokeTest;
+
+/**
+ * @group queue_driver_smoke-test
+ */
+class QueueDriverConnectionSmokeTestTest extends TestCase
+{
+    /**
+     * Test that the smoke test can be test...ed
+     */
+    public function testHappyPath()
+    {
+        $queueDriver = $this->createMock(QueueDriverInterface::class);
+        $queueDriver
+            ->method('isConnected')
+            ->willReturn(true);
+        $queueDriver
+            ->method('send')
+            ->willReturn(true);
+
+        $test = new QueueDriverConnectionSmokeTest($queueDriver);
+        $result = $test->run();
+
+        $this->assertInstanceOf(SmokeTestOutput::class, $result, sprintf('Expected a SmokeTestOutput instance as result, got %s instead', get_class($result)));
+        $this->assertTrue($result->isOK(), 'Connection failed on smoke test');
+    }
+
+    /**
+     * Test connection failures. Should catch anything.
+     */
+    public function testConnectionFailure()
+    {
+        $queueDriver = $this->createMock(QueueDriverInterface::class);
+        $queueDriver->expects($this->once())
+            ->method('connect')
+            ->willThrowException(new class('LiteraryAnyException') extends \Exception {
```

Comment:
I think here we could just throw \Exception rather than creating a class on the fly

Reply:
Done.